tests/TileDB.CSharp.Test/DeletesTest.cs

```diff
@@ -0,0 +1,896 @@
+
+            // Check for expected values
+            var readBuffers = new Dictionary<string, System.Array>()
+                { { "rows", new int[16] }, { "cols", new int[16] }, { "a1", new int[16] } };
+            using var readQuery = TestUtil.ReadArray(array, layout, readBuffers, ctx: ctx);
+
+            array.Open(QueryType.Read);
+            Assert.AreEqual(15UL, readQuery.ResultBufferElements()["a1"].Item1);
+            array.Close();
+            var expectedData = new Dictionary<string, System.Array>()
+            {
+                { "a1", new[] { 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 0 } },
+                { "rows", new[] { 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4, 0 } },
+                { "cols", new[] { 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4, 0 } },
+            };
+            TestUtil.CompareBuffers(expectedData, readBuffers, duplicates);
+        }
+
+        [DataRow(LayoutType.Unordered, "legacy", true)]
+        [DataRow(LayoutType.Unordered, "legacy", false)]
+        [DataRow(LayoutType.Unordered, "refactored", true)]
+        [DataRow(LayoutType.Unordered, "refactored", false)]
+        [DataRow(LayoutType.RowMajor, "legacy", true)]
+        [DataRow(LayoutType.RowMajor, "legacy", false)]
+        [DataRow(LayoutType.RowMajor, "refactored", true)]
+        [DataRow(LayoutType.RowMajor, "refactored", false)]
+        [DataRow(LayoutType.GlobalOrder, "legacy", true)]
+        [DataRow(LayoutType.GlobalOrder, "legacy", false)]
+        [DataRow(LayoutType.GlobalOrder, "refactored", true)]
+        [DataRow(LayoutType.GlobalOrder, "refactored", false)]
+        [DataTestMethod]
+        public void VaryingCells(LayoutType layout, string queryReader, bool duplicates)
+        {
+            string arrayName = "deletes-varying-cells";
+            var ctx = Context.GetDefault();
+            ctx.Config().Set("sm.query.sparse_global_order.reader", queryReader);
+            ctx.Config().Set("sm.query.sparse_unordered_with_dups.reader", queryReader);
+            InitTest(ctx, arrayName, layout, duplicates);
+            using var array = new Array(ctx, arrayName);
+
+            DeleteFour(ctx, array, layout, duplicates);
+
+            // Delete 1 cell
+            array.Open(QueryType.Delete);
+            using var deleteQuery = new Query(ctx, array, QueryType.Delete);
+            using var colsCondition = new QueryCondition(ctx);
+            colsCondition.Init("cols", 1, QueryConditionOperatorType.Equal);
+            using var rowsCondition = new QueryCondition(ctx);
+            rowsCondition.Init("rows", 1, QueryConditionOperatorType.Equal);
+            using var andCondition = colsCondition.Combine(rowsCondition, QueryConditionCombinationOperatorType.And);
+            deleteQuery.SetCondition(andCondition);
+            deleteQuery.Submit();
+            array.Close();
+            Assert.AreEqual(deleteQuery.Status(), QueryStatus.Completed);
```

Comment:
The order of parameters must be swapped; first goes the expected value and then the actual. It should be caught by SonarSource; opened SonarSource/sonar-dotnet#6547.

Reply:
I swapped the order of these assertions but it looks like the NOT operator isn't supported yet for [QueryConditionCombinationOp](https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/query/query_condition.cc#L2122)